test/models/undeployed_commits_test.rb

```diff
@@ -39,6 +39,48 @@ class UndeployedCommitsTest < ActiveSupport::TestCase
       assert_predicate @commit, :deploy_discouraged?
     end
 
+    test "#deploy_scheduled? returns true if the stack has continuous deployment enabled, commit index is lower then the maximum commits per deploy and commit is not part of active task" do
+      commit = UndeployedCommit.new(shipit_commits(:undeployed_4), 1)
+      active_task = commit.stack.active_task
+
+      assert_predicate commit.stack, :continuous_deployment
+      assert_equal 1, commit.index
+      assert_equal 3, commit.stack.maximum_commits_per_deploy
+      refute commit.id > active_task.since_commit.id && commit.id <= active_task.until_commit.id
```

Comment:
This seems to be duplicating the logic of `active?`, so `refute_predicate commit, active?` would be much tidier.